cache.js

```diff
@@ -1,77 +1,118 @@
- *
- * @memberof cache
- * @param {string} key key whose mapping is to be checked in the map
- * @returns {boolean} whether the key has a mapping
- */
-const exists = function (key) {
-  return get(key) !== null;
-};
+  /**
+   * Returns the value to which the specified key is mapped.
+   *
+   * @param {string} key the key whose associated value is to be returned
+   * @param {function} [defaultSupplier] if the specified key is not already associated with a value, this function will return a default value
+   * @returns {*|null} the current object for the supplied key, a default value if defaultSupplier is provided, or null
+   */
+  get (key, defaultSupplier) {
+    if (this._deprecated === true) logDeprecationWarning('get');
+    if (typeof defaultSupplier === 'function') {
+      return this._valueCache.get(key, defaultSupplier);
+    } else {
+      return this._valueCache.get(key);
+    }
+  }
+
+  /**
+   * Associates the specified value with the specified key.
+   *
+   * @param {string} key key with which the specified value is to be associated
+   * @param {*} value value to be associated with the specified key
+   * @returns {*|null} the previous value associated with the key, or null if there was no mapping for key
+   */
+  put (key, value) {
+    if (this._deprecated === true) logDeprecationWarning('put');
+    return this._valueCache.put(key, value);
+  }
+
+  /**
+   * Removes the mapping for a key from this map if it is present.
+   *
+   * @param {string} key key whose mapping is to be removed from the cache
+   * @returns {*|null} the previous value associated with the key or null if there was no mapping for key
+   */
+  remove (key) {
+    if (this._deprecated === true) logDeprecationWarning('remove');
+    return this._valueCache.remove(key);
+  }
+
+  /**
+   * Checks the mapping for a key from this map.
+   *
+   * @param {string} key key whose mapping is to be checked in the map
+   * @returns {boolean} whether the key has a mapping
+   */
+  exists (key) {
+    if (this._deprecated === true) logDeprecationWarning('exists');
+    return this._valueCache.get(key) !== null;
+  }
+}
+
+let addonSharedJSCache;
+if (coreCacheAvail === true) {
+  log.debug('Caches from core are available, enable legacy cache methods to keep the old API.');
+  addonSharedJSCache = new JSCache(sharedCache, true);
+} else {
+  log.debug('Caches from core are unavailable, using the addon-provided cache.');
+  addonSharedJSCache = new JSCache(addonSharedCache);
+}
 
 module.exports = {
-  get,
-  put,
-  remove,
-  exists
+  /** @deprecated */
+  get: (key, defaultSupplier) => { return addonSharedJSCache.get(key, defaultSupplier); },
+  /** @deprecated */
+  put: (key, value) => { return addonSharedJSCache.put(key, value); },
+  /** @deprecated */
+  remove: (key) => { return addonSharedJSCache.remove(key); },
+  /** @deprecated */
+  exists: (key) => { return addonSharedJSCache.exists(key); },
+  /**
+   * Private cache for each script.
```

Comment:
The comments for `shared` and `private` are swapped.

Reply:
Thx, I fixed this.